language/move-binary-format/src/proptest_types/types.rs

```diff
@@ -188,15 +196,20 @@ impl StructDefinitionGen {
             .fold(self.abilities.materialize(), |acc, field| {
                 acc.intersect(state.potential_abilities(&field.signature.0))
             });
+
+        let type_parameters = self
+            .type_parameters
+            .into_iter()
+            .map(|(constraints, is_phantom)| StructTypeParameter {
+                constraints: constraints.materialize(),
+                is_phantom,
```

Comment:
Does this work? that is, can fields get generated with the type parameter as a type?

If so, you will have to consider that. 
If not, we should add that feature at some point

Reply:
Oh, didn't think of that, but yeah apparently it works. The field generator only generates concrete types as I see now.